Toolkit/ArcGISToolkit/TemplatePickerViewController.swift

```diff
@@ -0,0 +1,296 @@
+                                return
+                            }
+                            
+                            // update info with swatch
+                            infos[index].swatch = image
+                            
+                            // reload index where that info currently is
+                            if let index = strongSelf.indexPathForInfo(info){
+                                strongSelf.tableView.reloadRows(at: [index], with: .automatic)
+                            }
+                        }
+                    }
+                }
+            }
+            
+        }
+        
+    }
+    
+    // MARK: TableView delegate/datasource methods
+    
+    public func numberOfSectionsInTableView(_ tableView: UITableView) -> Int{
+        return tables.count
+    }
+    
+    public func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+        return tables.count == 0 ? nil : tables[section].tableName
+    }
+    
+    public func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
+        
+        // when the user taps on a feature type
+        
+        // first get the selected object
+        let selectedFeatureTemplateInfo = infoForIndexPath(indexPath)!
+        
+        // If the search controller is still active, the delegate will not be
+        // able to dismiss us, if desired.
+        // Note: we can't do this before the above, or else the object we pull
+        // out of the datasource will be incorrect
+        navigationItem.searchController?.isActive = false
+        
+        delegate?.templatePickerViewControllerDidSelectTemplate(self, featureTemplateInfo: selectedFeatureTemplateInfo)
+    }
+    
+    override public func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        
+        guard tables.count > 0 else {
+            return 0
+        }
+        
+        let tableName = tables[section].tableName
+        let infos = self.currentDatasource[tableName]
+        return infos?.count ?? 0
+    }
+    
+    override public func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(withIdentifier: cellReuseIdentifier, for: indexPath)
+        let info = infoForIndexPath(indexPath)!
+        cell.textLabel?.text = info.featureTemplate.name
+        cell.imageView?.image = info.swatch
+        return cell
+    }
+    
+    // MARK: go back, cancel methods
+    
+    @objc private func cancelAction(){
+        // If the search controller is still active, the delegate will not be
+        // able to dismiss us, if desired.
+        navigationItem.searchController?.isActive = false
+        delegate?.templatePickerViewControllerDidCancel(self)
+    }
+    
+    // MARK: IndexPath -> Info
+    
+    private func infoForIndexPath(_ indexPath: IndexPath) -> FeatureTemplateInfo?{
```

Comment:
In both of the places this is called, the result is force-unwrapped.  Can this be modified to prevent the force-unwrapping, either here or in the calling locations?

Reply:
good call